R/select_rows.r

```diff
@@ -17,12 +20,14 @@
 #' select_rows(iris, c(TRUE,FALSE)) # Select only odd rows
 #' iris2 <- iris; rownames(iris2) <- paste0("row", 1:nrow(iris2))
 #' select_rows(iris, c("row10", "row51")) # Select rows by name
-select_rows <- function(dataframe, rows, whole = FALSE, ...) {
+select_rows <- function(dataframe, rows, whole = FALSE, cols_to_check = NULL, ...) {
   force(rows); force(whole); args <- force(list(...))
+  cols <- names(dataframe)
+  if (!is.null(cols_to_check)) cols <- intersect(cols, cols_to_check)
   eval(substitute({
     dataframe <- dataframe[if(is.function(rows)) {
-      if (whole) rows(dataframe) else apply(dataframe, 1, rows)
-    } else rows, ]
+      if (whole) rows(dataframe[, cols, drop = FALSE]) else apply(dataframe[, cols, drop = FALSE], 1, rows)
+    } else rows, , drop = FALSE]
```

Comment:
nope, it's nested expressions. Very nasty, I agree, but that's how it was written originally :trollface: @robertzk